<launch>
    <!-- pylon_camera: pylon_camera_node -->
    <arg unless="false" name="launch_prefix" value="" />
    <arg     if="false" name="launch_prefix" value="gdb -ex run --args" />
    <node name="pylon_camera_node" pkg="pylon_camera" type="pylon_camera_node" required="true" output="screen"
        respawn="false" launch-prefix="$(arg launch_prefix)">
        <rosparam command="load" file="$(find pylon_camera)/config/default.yaml" />
    </node>

    <!-- mg400_bringup: mg400_bringup.launch -->
    <arg name="robot_ip" doc="IP of the controller" default="192.168.5.1"/>
    <param name="robot_description" textfile="$(find mg400_description)/urdf/mg400_description.urdf"/>
    <node name="mg400_bringup" pkg="mg400_bringup" type="mg400_bringup"  required="true" output="screen">
        <param name="joint_publish_rate" type="double" value="10"/>
        <param name="trajectory_duration" type="double" value="0.3"/>
        <param name="robot_ip_address" type="str" value="192.168.1.6"/>
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher"  required="true" type="robot_state_publisher"/>
    <node name="rviz" pkg="rviz" type="rviz"  required="true" args="-d $(find mg400_description)/rviz/urdf.rviz"/>

    <!-- softhand_mg400_img_control -->
    <node name="soft_img_control" pkg="softhand_mg400_img_control" type="soft_img_control" output="screen">
        <rosparam command="load" file="$(find softhand_mg400_img_control)/config/param.yaml" />
    </node>

</launch>